spec/services/solidus_bolt/transaction/capture_spec.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+RSpec.describe SolidusBolt::Transaction::Capture do
+  subject(:api) { described_class.new(transaction_reference: reference, amount: amount, currency: currency) }
+
+  let(:order) { create(:order) }
+  let(:reference) { 'Bolt_Reference_ID' }
+  let(:amount) { order.total }
+  let(:currency) { order.currency }
+
+  before do
+    create(:bolt_configuration)
+  end
+
+  describe '#call' do
+    let(:url) { "#{api.api_base_url}/#{api.api_version}/merchant/transactions/capture" }
+    let(:response) { instance_double(HTTParty::Response, body: response_body) }
+    let(:response_body) { 'Iron Man' }
+    let(:api_key) { SolidusBolt::BoltConfiguration.fetch.api_key }
+    let(:nonce) { '123456789012345' }
```

Comment:
Considering that generate_nonce is a method into the Base object, what do you think to create a shared_example to test that it's working?

it can be something like:

```
RSpec.shared_example 'unique nonce' do
  before {  allow(SecureRandom).to receive(hex).and_return('123456789012345') }
  
  it 'generate a new nonce' do
    expect(HTTParty).to have_received(:post).with(hash_including(headers: {'x-Nonce'}))
  end
end
```

Or something like this.

Reply:
I'm sorry, I couldn't understand this